Build.proj

```diff
@@ -1,17 +1,17 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
+
   <PropertyGroup>
     <BuildFlavour Condition=" '$(BuildFlavour)' == '' ">Debug</BuildFlavour>
-    <Mono Condition="'$(OS)' != 'Windows_NT'">true</Mono>
+    <Mono Condition="'$(MSBuildRuntimeType)' == 'Mono'">true</Mono>
```

Comment:
Not sure about this change. If `$(OS)` is not `Windows_NT`, it doesn't necessarily mean it's running on Mono, but is it possible that `$(MSBuildRuntimeType)` is not `Mono`, when building for Mono?

Reply:
Would be nice to not have a compile time Mono check and just do it at runtime.

Reply:
Agreed, it shouldn't be that difficult to do.

Reply:
It looks like the `Mono` property is defined here too: https://github.com/IronLanguages/dlr/blob/master/Build/Common.proj#L10.

The `Mono` property is currently used to define a flag which is used in one place only: https://github.com/IronLanguages/dlr/blob/340682806d1cd54d72147c7f2f8f5a120c1ba6be/Src/Microsoft.Dynamic/Interpreter/InterpretedFrame.cs#L139-L143
You mean checking at runtime like [this](https://stackoverflow.com/a/721194/4647866)?

Reply:
Yes, that type runtime check.

Reply:
We have a check in IronPython right now [here](https://github.com/IronLanguages/ironpython2/blob/40f51b92f8294bb901d29738707103f92fe19c87/Src/IronPython/Runtime/ClrModule.cs#L67). Honestly, it would be nice to migrate all the feature checking into one place, but I haven't thought about how I really want to do that yet.

Reply:
Should I revert the change on this line or move the Mono check to runtime?

Reply:
Let's revert for now and add something as a runtime check in a different PR